mllib/src/main/scala/org/apache/spark/ml/classification/DecisionTreeClassifier.scala

```diff
@@ -126,6 +126,8 @@ class DecisionTreeClassifier @Since("1.4.0") (
     }
     validateNumClasses(numClasses)
     val instances = extractInstances(dataset, numClasses)
+    val weightSum = instances.map(_.weight).reduce(_ + _)
```

Comment:
Hm, on second thought, this requires an extra pass to compute the weights. Is it worth it? is there any other place to get this info instead without adding a Spark job?

Reply:
I will take a look to find out if ```weightSum``` is already available in other place.